go/Dockerfile.build

```diff
@@ -0,0 +1,10 @@
+FROM golang:1.8-alpine
+RUN apk update && apk add --no-cache make git
+ENV GOPATH=/go
+ENV PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+
+# The project sources
+VOLUME ["/go/src/github.com/docker/hyperkit/go"]
+WORKDIR /go/src/github.com/docker/hyperkit/go
+
+ENTRYPOINT ["make"]
```

Comment:
I missed the `GOOS=darwin` in the `Makefile`. I think we also do need to support building without docker (i.e. with things in the correct places in `$GOPATH`). Reading the `Makefile` it looks like that should work if you call the correct `Makefile` target.